packages/grid/data-grid/src/tests/filtering.DataGrid.test.tsx

```diff
@@ -203,10 +196,10 @@ describe('<DataGrid /> - Filter', () => {
       { operator: 'before', value: new Date(2001, 0, 1), expected: ['12/1/2000'] },
       { operator: 'onOrBefore', value: new Date(2001, 0, 1), expected: ['12/1/2000', '1/1/2001'] },
     ].forEach(({ operator, value, expected }) => {
-      it(`should allow object as value and work with valueGetter, operator: ${operator}`, function dateOpsTest() {
+      it(`should allow object as value and work with valueGetter, operator: ${operator}`, () => {
         render(
           <TestCase
-            value={value.toLocaleDateString()}
+            value={value}
```

Comment:
Actually, @m4theushw you were right, this shouldn't be a Date object but a string. Otherwise, we are off the actual behavior of the component.

Reply:
I returned the string value in https://github.com/mui-org/material-ui-x/pull/1722 to match the actual implementation.